File: level0/level0.py
#!/usr/bin/env python

import sys

with open(sys.argv[1]) as f:
    content = frozenset(f.read().splitlines())

for line in sys.stdin.read().splitlines():
    res = []
    if line:
        for word in line.split(' '):
            if word.lower() in content:
                res.append(word)
            else:
                res.append('<{0}>'.format(word))
        print ' '.join(res)
    else:
        print

File: level0/level0.pl
#!/usr/bin/perl

open(my $f, $ARGV[0]);
chomp(@w = <$f>);
@h{@w} = map(1, @w);
sub ff {
    $v = $_[0];
    if ($v =~ /\W/) {
        "<$v>"
    } else {
        if (exists($index{$v})) {
            $index{$v}
        } else {
            $index{$v} = $h{lc($v)} ? $v : "<$v>";
        }
    }
}
while (<>) {
    s/([^ \n]+)/&ff($+)/eg;
}
